src/query.c

```diff
@@ -521,11 +517,7 @@ QueryIterator *Query_EvalTokenNode(QueryEvalCtx *q, QueryNode *qn) {
   const FieldSpec *fs = IndexSpec_GetFieldByBit(q->sctx->spec, qn->opts.fieldMask);
   RSQueryTerm *term = NewQueryTerm(&qn->tn, q->tokenId++);
 
-  QueryIterator *it = Redis_OpenReader(q->sctx, term, q->docTable, EFFECTIVE_FIELDMASK(q, qn), qn->opts.weight);
-  if (!it) {
-    Term_Free(term);
-  }
-  return it;
+  return Redis_OpenReader(q->sctx, term, q->docTable, EFFECTIVE_FIELDMASK(q, qn), qn->opts.weight);
```

Comment:
OpenReader cannot return Null now? or is it about ownership?

Reply:
It's now consuming the `term`, even on failure

Reply:
I see now